mm2src/adex_cli/src/scenarios/init_mm2_cfg.rs

```diff
@@ -0,0 +1,302 @@
+
+        self.rpc_password = Text::new("What is the rpc_password:")
+            .with_help_message("Your password for protected RPC methods (userpass)")
+            .with_validator(validator)
+            .with_default(default_password.as_str())
+            .with_placeholder(default_password.as_str())
+            .prompt()
+            .map_err(|error| {
+                error!("Failed to get rpc_password: {error}");
+            })?
+            .into();
+        Ok(())
+    }
+
+    fn generate_password() -> Result<String, ()> {
+        let pg = PasswordGenerator {
+            length: 8,
+            numbers: true,
+            lowercase_letters: true,
+            uppercase_letters: true,
+            symbols: true,
+            spaces: false,
+            exclude_similar_characters: false,
+            strict: true,
+        };
+        pg.generate_one()
+            .map_err(|error| error!("Failed to generate password: {error}"))
+    }
+
+    fn inquire_allow_weak_password(&mut self) -> Result<(), ()> {
+        self.allow_weak_password = Confirm::new("Allow weak password:")
+                .with_default(false)
+                .with_placeholder("No")
+                .with_help_message(r#"If true, will allow low entropy rpc_password. If false rpc_password must not have 3 of the same characters in a row, must be between 8-32 characters in length, must contain at least one of each of the following: numeric, uppercase, lowercase, special character (e.g. !#$*). It also can not contain the word "password", or the chars <, >, and &. Defaults to false."#)
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get allow_weak_password: {error}");
+                })?
+                .into();
+        Ok(())
+    }
+
+    fn inquire_userhome(&mut self) -> Result<(), ()> {
+        self.userhome = CustomType::<InquireOption<String>>::new("What is userhome:")
+                .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+                .with_help_message(r#"The path to your home, called from your environment variables and entered as a regular expression. Example: /${HOME#"/"}"#)
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get userhome: {error}");
+                })?.into();
+        Ok(())
+    }
+
+    fn inquire_rpcip(&mut self) -> Result<(), ()> {
+        self.rpcip = CustomType::<InquireOption<Ipv4Addr>>::new("What is rpcip:")
+            .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+            .with_help_message("IP address to bind to for RPC server. Optional, defaults to 127.0.0.1")
+            .prompt()
+            .map_err(|error| {
+                error!("Failed to get rpcip: {error}");
+            })?
+            .into();
+        Ok(())
+    }
+
+    fn inquire_rpcport(&mut self) -> Result<(), ()> {
+        self.rpcport = CustomType::<InquireOption<u16>>::new("What is the rpcport:")
```

Comment:
Default port and some validations
https://github.com/KomodoPlatform/atomicDEX-API/blob/59e5c786c30a6abf068704791700dbcad9da7b46/mm2src/mm2_core/src/mm_ctx.rs#L182-L188

Reply:
The process was performed in accordance with the rules set by IANA (Internet Assigned Numbers Authority). Currently, the 'rpcport' is being checked to see if it falls within the range of [1024, 49152].

done